metriq_gym/run.py

```diff
@@ -102,7 +102,10 @@ def poll_job(args: argparse.Namespace, job_manager: JobManager) -> None:
     ]
     if all(task.status() == JobStatus.COMPLETED for task in quantum_jobs):
         result_data: list[GateModelResultData] = [task.result().data for task in quantum_jobs]
-        print(handler.poll_handler(job_data, result_data, quantum_jobs))
+        results: BenchmarkResult = handler.poll_handler(job_data, result_data, quantum_jobs)
+        print(results)
+        if is_upload:
+            handler.upload_handler(job_data, results, metriq_job.dispatch_time, args.submission_id, platforms[metriq_job.device_name.lower()])
```

Comment:
This lookup will fail when new devices are introduced, as they come dynamically from the provider APIs. For example, let's say aws introduces a new device with string `arn:aws:braket:eu-north-1::device/fancy/qpu`, then we won't have it in the `plaforms` dict. We could either create the device in the backend if it doesn't exist, or support only a limited set of devices.